### PR TITLE
Feat 4392 add code analyzer

### LINKED ISSUE #4392: 🔎 Add Code Analyzer To ‘utopia-php/preloader’
# 💭 Introduction

Code analyzer is one of the most valuable tools in the open source world. It not only ensures every high quality of the code but also allows CI/CD to ensure this. A code analyzer can find a potential bug, unexpected null value, or type mismatch. Having a code analyzer set up should be a high priority for every project as it makes the project more stable and bug-proof.

Your task is to update our library [utopia-php/preloader](https://github.com/utopia-php/preloader) and implement a code analyzer. We would like to use [PHPStan](https://phpstan.org/) as our analyzer.

**To make this task easy, we created a sample pull request showing what exactly needs to be done to implement the analyzer into the Utopia PHP library: [https://github.com/utopia-php/ab/pull/8](https://github.com/utopia-php/ab/pull/8)**

### 🎯 Requirements

- Experience with GitHub Actions.
- Experience with Composer.
- Experience with PHP.

### ✅ Task Summary

- [ ]  Ask to be assigned to the issue.
- [ ]  Wait to be assigned.
- [ ]  Implement code analyzer in [utopia-php/preloader](https://github.com/utopia-php/preloader) (use [https://github.com/utopia-php/ab/pull/8](https://github.com/utopia-php/ab/pull/8) as template)
- [ ]  Run code analyzer to check existing code, and update it if required
- [ ]  Submit a pull request in [utopia-php/preloader](https://github.com/utopia-php/preloader).

> Please configure PHPStan to max level (9). If that gives you too many problems that you cannot solve, please drop to level 8. If you have trouble with level 8 as well, let us know, and we can consider a lower level on some repositories.

If you have questions, need any help, or just want to hang out, make sure to join us on our [Discord server](https://appwrite.io/discord).

**Happy Appwriting!**

### PR DESCRIPTION
**Pull request by maxslimb**
Implemented code analyzer PHPStan, and updated the code quality.

### CODE REVIEW

**Comment by Meldiron** [comment]
@maxslimb , thanks for the PR! 🤯 Please give us some time to review it.

**Comment by maxslimb** [comment]
@christyjacob4 please review and merge this commit!
Thank You!

**Review by stnguyen90 — DISMISSED** [review]
Great PR! 🤯 Would you please fix the merge conflicts?

**Comment by PineappleIOnic** [comment]
Thank you so much for the PR 🤩. We're adding the `hacktoberfest-accepted` label to ensure this PR counts towards your Hacktoberfest contributions count. With that said, please stay active on this PR to address any comments once you receive a review. Happy Hacktoberfest! 🎃

File: composer.json [hunk 1]
```diff
@@ -12,12 +17,16 @@
...
     },
     "require": {
-        "php": ">=7.4"
+        "php": ">=8.0"
     },
     "require-dev": {
         "phpunit/phpunit": "^9.3",
         "vimeo/psalm": "4.0.1",
+
+        "phpstan/phpstan": "1.9.x-dev"
```
Comment (stnguyen90) [inline composer.json:26]: There's a problem with this JSON

**Comment by stnguyen90** [comment]
Please make sure to mark the PR ready for review and re-request a review when you're done with your changes

**Review by christyjacob4 — APPROVED** [review]

**Comment by christyjacob4** [comment]
THANK YOU! All changes merged 🥳

Please reach out to me on our [Discord server](https://appwrite.io/discord) if you would like to claim your Appwrite swags! As a way of saying thank you, we would also love to invite you to join the Appwrite organization on GitHub. Please share your GitHub username with us on Discord.  

You can accept the invite by visiting https://github.com/orgs/appwrite/invitation. By joining our team, you will officially be an Appwrite maintainer on GitHub.

You can change your membership [visibility settings](https://help.github.com/en/github/setting-up-and-managing-your-github-user-account/publicizing-or-hiding-organization-membership), so your new Appwrite team membership badge will show up on your personal GitHub profile.

Please feel free to look for more PRs you might be interested in helping with on our long list of [Hacktoberfest friendly issues](https://github.com/search?q=org%3Aappwrite+org%3Autopia-php+org%3Aopen-runtimes+label%3Ahacktoberfest+created%3A%3E2021-01-01+state%3Aopen&type=issues&s=updated&o=asc) and help make Appwrite better :)

**Pull request merged**